config/dynamic-config-builder/src/bin/full-node-config-builder.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use dynamic_config_builder::FullNodeConfig;
+use libra_config::config::{NodeConfig, PersistableConfig};
+use parity_multiaddr::Multiaddr;
+use std::{convert::TryInto, fs, path::PathBuf};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(about = "Tool to create Libra Validator Configs")]
+struct Args {
+    // Describe the validator networrk
+    #[structopt(short = "n", long, default_value = "1")]
+    /// Specify the number of nodes to configure
+    nodes: usize,
+    #[structopt(short = "s", long)]
+    /// Use the provided seed for generating keys for each of the validators
+    seed: Option<String>,
+
+    // Parameters for this full node config
+    #[structopt(short = "a", long, parse(from_str = parse_addr))]
+    /// Advertised address for this node, if this is null, listen is reused
+    advertised: Multiaddr,
+    #[structopt(short = "b", long, parse(from_str = parse_addr))]
+    /// Advertised address for the first node in this test net
+    bootstrap: Multiaddr,
+    #[structopt(short = "d", long, parse(from_os_str))]
+    /// The data directory for the configs (e.g. /opt/libra/etc)
+    data_dir: PathBuf,
+    #[structopt(short = "c", long)]
+    /// Use the provided seed for generating keys for each of the FullNodes
+    full_node_seed: Option<String>,
+    #[structopt(short = "f", long, default_value = "1")]
+    /// Total number of full nodes
+    full_nodes: usize,
+    #[structopt(short = "i", long, default_value = "0")]
+    /// Specify the index into the number of nodes to write to output dir
+    index: usize,
+    #[structopt(short = "l", long, parse(from_str = parse_addr))]
+    /// Listening address for this node
+    listen: Multiaddr,
+    #[structopt(short = "o", long, parse(from_os_str))]
+    /// The output directory, note if a config exists already here, it will be updated to include
+    /// this full node network
+    output_dir: PathBuf,
+    #[structopt(short = "p", long)]
+    /// Public network, doesn't use any authentication or encryption
+    public: bool,
+    #[structopt(short = "t", long, parse(from_os_str))]
+    /// Path to a template NodeConfig
+    template: Option<PathBuf>,
+}
+
+fn parse_addr(src: &str) -> Multiaddr {
+    src.parse::<Multiaddr>().unwrap()
```

Comment:
maybe change the unwrap to `expect("address must be correctly formatted as multiaddr")`